### PR TITLE
Add support for Mongo 4.0

### PR DESCRIPTION
**Pull request by UserNotFound**
Mongo 4.0 drops support for TLS v1.0, so tests are included for supported protocols.

Installation source is not present in Debian Wheezy (EOL'ed), so an upgrade to Jessie is also included.

Minor changes to some `docker` commands in tests to ensure no stopped containers are left on the system. Also adds `start_mongodb()` to keep tests DRY.

### CODE REVIEW

File: Dockerfile.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM quay.io/aptible/debian:wheezy
+FROM quay.io/aptible/debian:jessie
```
Comment (krallin) [inline Dockerfile.erb]: Can we go all the way to stretch while we're in there (we needed to do that in MySQL already - let's try and standardize on stretch).

File: .travis.yml [hunk 2]
```diff
@@ -10,6 +10,7 @@ env:
     - TAG=3.2
     - TAG=3.4
     - TAG=3.6
+    - TAG=4.0
```
Comment (krallin) [inline .travis.yml]: Update `latest.mk` as well

File: test/ssl-support.bats [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+local_s_client() {
+  echo OK | openssl s_client -connect localhost:27017 "$@"
+}
+
+@test "It should allow connections using TLS1.2" {
+  start_mongodb
+
+  run local_s_client -tls1_2
+  [ "$status" -eq 0 ]
```
Comment (krallin) [inline test/ssl-support.bats]: Same comments as in https://github.com/aptible/docker-influxdb/pull/6 regarding `run` here

File: test/ssl-support.bats [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+@test "It should allow connections using TLS1.1" {
+  start_mongodb
+
+  run local_s_client -tls1_1
+  [ "$status" -eq 0 ]
+}
+
+@test "It should allow connections using TLS1.0" {
```
Comment (krallin) [inline test/ssl-support.bats]: Can you rename the test a little bit to clarify what version we are testing?

File: test/ssl-support.bats [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  if dpkg --compare-versions "$MONGO_VERSION" ge 4; then
+    skip "MongoDB 4.0+ does not support TLS1.0"
```
Comment (krallin) [inline test/ssl-support.bats]: Call it `MongoDB >=4.0` to be consistent with yourself in the next test :)

File: Dockerfile.erb (hunk 1 above)
Comment (UserNotFound) [reply Dockerfile.erb]: It looks like 2.6, 3.2, and 3.4 don't have Stretch repositories available - I left these on Jessie and updated 3.6 and 4.0 to Stretch.

File: test/ssl-support.bats (hunk 4 above)
Comment (UserNotFound) [reply test/ssl-support.bats]: Is this more clear?

```
 ✓ MongoDB <4.0 should allow connections using TLS1.0
 - MongoDB >=4.0 should disallow connections using TLS1.0 (skipped: MongoDB 3.2.19 does support TLS1.0)
```
and
```
 - MongoDB <4.0 should allow connections using TLS1.0 (skipped: MongoDB 4.0.0 does not support TLS1.0)
 ✓ MongoDB >=4.0 should disallow connections using TLS1.0
```

Comment (krallin) [reply test/ssl-support.bats]: Yeah that's good :+1:

File: 3.2/config.mk [hunk 6]
```diff
@@ -3,3 +3,5 @@ export MONGO_SHA1SUM = NOT_USED_INSTALLED_FROM_PACKAGES
 export MONGO_GPG_KEY = EA312927
 
 export MONGO_SSL_MODE = requireSSL
+
+export DEBIAN_VERSION = jessie
```
Comment (krallin) [inline 3.2/config.mk]: Why you gotta hate ending files with a newline :cry:

**Pull request merged**